test/validation/api/dma/dma.c

```diff
@@ -636,6 +641,64 @@ static void test_dma_addr_to_addr_trs(odp_dma_compl_mode_t compl_mode_mask, uint
 	CU_ASSERT(odp_dma_destroy(dma) == 0);
 }
 
+static void test_dma_addr_to_addr_max_trs(odp_dma_compl_mode_t compl_mode_mask)
+{
+	odp_dma_param_t dma_param;
+	uint32_t num_trs = global.dma_capa.max_transfers;
+	odp_dma_transfer_param_t trs_param[num_trs];
+	odp_dma_t dma;
+	odp_dma_seg_t src_seg[num_trs];
+	odp_dma_seg_t dst_seg[num_trs];
+	int ret;
+	uint32_t i, cur_len;
+	uint8_t *src = global.src_addr + OFFSET;
+	uint8_t *dst = global.dst_addr + OFFSET;
+	uint32_t len = global.len;
+	uint32_t seg_len = len / num_trs;
```

Comment:
Depending on `max_transfers` capa this could theoretically round down to zero.

Reply:
Yes, could be safer to incorporate `max_transfers` to the SHM calculation. The size will skyrocket though if the implementation is a capable one.

Reply:
Added a new patch for the max segment length stuff in v2, hopefully now properly taken into account (3069d430c8a50d783b5a52f88fc84680aef3fd7a).